frontend/src/components/Signup.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch, batch } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import Swal from 'sweetalert2'
+
+import user from '../reducers/user'
+
+import { API_URL } from '../reusable/urls'
+
+const Signup = () => {
```

Comment:
both your forms look 99% the same, maybe it is a good idea to create one and pass differences via props?